das-cli/src/common/utils.py

```
@@ -195,5 +195,20 @@ def extract_service_port(endpoint: str) -> int | None:
         return None
 
 
+def require_endpoint_port(endpoint: Any, setting_name: str) -> int:
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `endpoint`

(ANN401)